File: ksh-java-template/src/main/java/io/gitlab/arturbosch/ksh/java/SettingsProvider.java
package io.gitlab.arturbosch.ksh.java;

import io.gitlab.arturbosch.ksh.api.ShellSettings;
import io.gitlab.arturbosch.ksh.api.provider.ShellSettingsProvider;
import io.gitlab.arturbosch.kutils.Container;
import org.jetbrains.annotations.NotNull;

import java.io.File;
import java.io.IOException;
import java.io.UncheckedIOException;

public class SettingsProvider implements ShellSettingsProvider {

    @Override
    public ShellSettings provide(@NotNull Container container) {
        try {
            String tmpFile = File.createTempFile("ksh-java-integration", null).toString();
            return new ShellSettings() {

                @Override
                public int getPriority() {
                    return 0;
                }

                @NotNull
                @Override
                public String getApplicationName() {
                    return "fromJava: ";
                }

                @NotNull
                @Override
                public String getHistoryFile() {
                    return tmpFile;
                }

                @NotNull
                @Override
                public String prompt() {
                    return getApplicationName();
                }
            };
        } catch (IOException e) {
            throw new UncheckedIOException(e);
        }
    }

    @Override
    public int getPriority() {
        return 0;
    }
}

File: ksh-template/src/main/kotlin/io/gitlab/arturbosch/ksh/template/SettingsProvider.kt
package io.gitlab.arturbosch.ksh.template

import io.gitlab.arturbosch.ksh.api.ShellSettings
import io.gitlab.arturbosch.ksh.api.provider.ShellSettingsProvider
import io.gitlab.arturbosch.kutils.Container
import java.nio.file.Files

class SettingsProvider : ShellSettingsProvider {

    override fun provide(container: Container): ShellSettings {
        return object : ShellSettings {
            override val applicationName: String = "app"
            override val historyFile: String = Files.createTempFile(applicationName, "history").toString()
            override fun prompt(): String = "$applicationName> "
        }
    }
}

